src/community/backup-restore/core/src/main/java/org/geoserver/backuprestore/processor/CatalogItemProcessor.java

```diff
@@ -0,0 +1,518 @@
+        // or the temporary one (Restore).
+        // 
+        // For restore operations the order matters.
+        JobExecution jobExecution = stepExecution.getJobExecution();
+        this.xstream = xStreamPersisterFactory.createXMLPersister();
+        if (backupFacade.getRestoreExecutions() != null
+                && !backupFacade.getRestoreExecutions().isEmpty()
+                && backupFacade.getRestoreExecutions().containsKey(jobExecution.getId())) {
+            this.currentJobExecution = backupFacade.getRestoreExecutions().get(jobExecution.getId());
+            this.catalog = ((RestoreExecutionAdapter)currentJobExecution).getRestoreCatalog();
+            this.isNew = true;
+        } else {
+            this.currentJobExecution = backupFacade.getBackupExecutions().get(jobExecution.getId());
+            this.catalog = backupFacade.getCatalog();
+            this.xstream.setExcludeIds();
+            this.isNew = false;
+        }
+        
+        Assert.notNull(this.catalog, "catalog must be set");
+
+        this.xstream.setCatalog(this.catalog);
+        this.xstream.setReferenceByName(true);
+        this.xp = this.xstream.getXStream();
+
+        Assert.notNull(this.xp, "xStream persister should not be NULL");
+        
+        JobParameters jobParameters = this.currentJobExecution.getJobParameters();
+        
+        this.dryRun = Boolean.parseBoolean(jobParameters.getString(Backup.PARAM_DRY_RUN_MODE, "false"));
+        this.bestEffort = Boolean.parseBoolean(jobParameters.getString(Backup.PARAM_BEST_EFFORT_MODE, "false"));
+    }
+
+    /**
+     * @return the xp
+     */
+    public XStream getXp() {
+        return xp;
+    }
+
+    /**
+     * @param xp the xp to set
+     */
+    public void setXp(XStream xp) {
+        this.xp = xp;
+    }
+
+    /**
+     * @return the isNew
+     */
+    public boolean isNew() {
+        return isNew;
+    }
+
+    /**
+     * @return the currentJobExecution
+     */
+    public AbstractExecutionAdapter getCurrentJobExecution() {
+        return currentJobExecution;
+    }
+
+    /**
+     * @return the dryRun
+     */
+    public boolean isDryRun() {
+        return dryRun;
+    }
+
+    /**
+     * @return the bestEffort
+     */
+    public boolean isBestEffort() {
+        return bestEffort;
+    }
+    
+    @Override
+    public T process(T resource) throws Exception {
+
+        if (resource != null) {
+            if (isNew) {
+                // Disabling additional validators
+                ((CatalogImpl) this.catalog).setExtendedValidation(false);
+            }
+            
+            LOGGER.info("Processing resource: " + resource + 
+                    " - Progress: [" + currentJobExecution.getExecutedSteps() + "/" + currentJobExecution.getTotalNumberOfSteps() + "]");
+
+            if (resource instanceof WorkspaceInfo) {
+                if (!validateWorkspace((WorkspaceInfo) resource, isNew)) {
```

Comment:
Hmm... if this happens, then we cannot restore the entire workspace and everything that's inside of it, or depending on ti. The same goes for the other resources stuff depends on (like, stores, layers vs groups containing them, and so on).
Just saying, having invalid stuff around is not uncommon, do we really want to lose it during a backup?
Also thinking about best effort vs fail fast mode, in the first case we should probably try to backup invalid, in the second fail I guess... even if if the backup is unattended, not having a backup at all might be a problem...